### PR TITLE
Add method to retrieve multiple deploymentConfigs

### PR DESCRIPTION
**Pull request by bart88**
Rather than retrieve by extact name, retrieve via label from which we can retrieve collections.

### CODE REVIEW

**Comment by pingers** [comment]
Would there be a time we want to retrieve by name? Maybe you can describe the scenario for labels with an example? Cheers

**Comment by bart88** [comment]
In our usage .. no. Because we are working with collections deployments tied to the concept of a site. Also the endpoint is for retrieving a collection of deployment configs binded by namespace. name was never used.

**Comment by singularo** [comment]
Retrieve by name will give an individual, by label gives a list, even if there is only one entry, so retrieval by name is better for being sure when deleting imho.

**Comment by singularo** [comment]
I'd prefer if the name param stayed, like the other api calls I've done, just makes things easier.
Deployments are per env, not site.

**Comment by bart88** [comment]
In lieu of the discussion about leaving the existing `getDeploymentConfig` method. I have added `getDeploymentConfigs` - for the retrieval of multiple deployment configs.

File: src/Client.php [hunk 1]
```diff
@@ -49,141 +49,147 @@ class Client implements OpenShiftClientInterface {
...
         'action' => 'PUT',
-        'uri'    => '/oapi/v1/namespaces/{namespace}/deploymentconfigs/{name}',
+        'uri' => '/oapi/v1/namespaces/{namespace}/deploymentconfigs/{name}',
       ],
     ],
-    'service'               => [
+    'deploymentconfigs' => [
+      'get' => [
+        'action' => 'GET',
+        'uri' => '/oapi/v1/namespaces/{namespace}/deploymentconfigs',
```
Comment (pingers) [inline src/Client.php]: The labels are parameters I assume

File: src/OpenShiftClientInterface.php [hunk 2]
```diff
@@ -444,6 +444,18 @@ public function updateDeploymentConfig(string $name, string $image_stream_tag, s
   public function deleteDeploymentConfig(string $name);
 
   /**
+   * Retrieve multiple deployment configs.
```
Comment (pingers) [inline src/OpenShiftClientInterface.php]: by label?

File: src/Client.php (hunk 1 above)
Comment (bart88) [inline src/Client.php]: Label is part of the query string. 

```
private function apiCall(string $method, $name = '', $label = NULL) {
    $resourceMethod = $this->getResourceMethod($method);
    $uri = $this->createRequestUri($resourceMethod['uri'], [
      'name' => (string) $name,
    ]);

    $query = [];
    if (!empty($label)) {
      $query = ['labelSelector' => $label];
    }

    return $this->request($resourceMethod['action'], $uri, [], $query);
  }
```

File: src/OpenShiftClientInterface.php (hunk 2 above)
Comment (bart88) [inline src/OpenShiftClientInterface.php]: 👍

**Comment by pingers** [comment]
:shipit:

**Pull request merged**